### PR TITLE
Refactor Code for OOP-SchoolLibrary app

### PR DESCRIPTION
**Pull request by enmanuelbayen**
# FEATURES

- Create new folders to put all class files besides `main.rb` to improve organization on the repo.
- Refactor the main method into a class with only the methods to print the menu and take user input.
- Update readme
- Check for linters error

### CODE REVIEW

File: classes/app.rb
Comment (Gambit142) [inline classes/app.rb:1]: - [ ] All codes responsible for receiving inputs or codes responsible for validating the input from the user should be put in the main.rb file or a separate class or class method or module. The other methods for creating a class instance or listing instances should only create those instances. 👍 😎 

---

For Example, the `create_book` method should only create a new instance of the `Book` class while codes for getting the `title` and `author` of the _**book**_  from the console app should be put in a separate method, class or module. 💻 

--- 

Kindly apply these changes for the following methods in the app.rb file:

- `create_person`
- `create_book`
- `create_rental`
- `list_rentals`

---

#### N.B: Please ensure that the app works as it should after refactoring the code. 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @enmanuelbayen @Ali-0111 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- README file is professional ✔️

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: classes/app.rb
Comment (enmanuelbayen) [reply classes/app.rb:1]: Thanks for the reply! Make the change by create a new class userInputHandler. This is now handling the user input for creating people, books, and rentals, while the App class is focused on its core responsibilities.

**Review by Davidosky007 — APPROVED** [review]
Hi @enmanuelbayen 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* No linter errors 👍 
* Ruby code refactored properly 👍 
* Well written PR 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**